### DISCUSSION TITLE
BeginPopup called only once

### DISCUSSION THREAD

**Discussion by SadE54** [ocornut/imgui#4671] [Old/Archived discussions]
Hi , 

I have a strange behavior with 2 popups. One can be displayed correctly  , and the other is only rendered once 
Here's a simplified version of this code part : 

```cpp
bool open_popup_1 = false;
bool open_popup_2 = false;
...
ImGui::Text("text);
if (ImGui::IsItemHovered() && ImGui::IsMouseDoubleClicked(0))
{
    open_popup_1 = true;
}
...
if (ImGui::Button("Button")
{
    open_popup_2 = true;
}
...

if (open_popup_1 == true)
{
    ImGui::OpenPopup("popup1");
}
if (open_popup_2 == true)
{
    ImGui::OpenPopup("popup2");
}

if (ImGui::BeginPopup("popup1"))
{
    ImGui::Text("Test 1");
    ImGui::EndPopup();
	
}

if (ImGui::BeginPopup("popup2"))
{
    ImGui::Text("Test 2");
    ImGui::EndPopup();
}
...
```
In both cases , the corresponding openpopup is triggered correctly. 
But only Popup 2 is rendered each frame once opened.
The ImGui::BeginPopup("popup1")) is called only once after ImGui::OpenPopup("popup1");

I really wondering what could cause such a behavior 🤔

PS: using 1.84 , DX9 , on win 10 x64.

**Comment by ocornut** [comment]
Hello,

Are you sure the the exact code you posted here would exhibit the bug? That would be very surprising.

**Reply by SadE54** [reply]
I did not try it directly , just reduced it to something similar to my base code . In fact I'm using pointer on objects  to trigger the openpopup , but this behavior is working similar to the bool way. I'm using node editor from thedmd, so it could be some interaction with it. 
That's why , primarily , as suggested by him i deferred the popup management outside the node rendering itself.
But with some luck I get it working each frame, using ImGuiWindowFlags_ChildWindow flag for the popup.
The only problem with this flag is that I cannot quit the popup clicking outside it.
But may be it can direct us to a start of explanation 🤨

**Comment by ocornut** [comment]
I am pretty sure the code you posted should work. This should be enough of a hint that we can’t help you if you don’t reduce your issues to the minimum amount of code that reproduce the issue and verify it. Doing so most of the time you would solve the issue yourself.

Also you posted on the wrong location and the right location (issues) gives you exactly those instructions.

**Comment by ocornut** [comment]
I confirm that the code you posted (after fixing the compile error) works perfectly.

**Comment by SadE54** [comment]
I understand, but I m in the case where I cannot post the complete code.
Even if my part of code is simple as I posted, I guess the problem is
linked to some interaction with other imgui based framework. Do you just
have an idea why the popup is working (almost) when the
imguiwindowflag_childwindow is used? Just to get a bone to chew 😊

**Reply by ocornut** [reply]
You don't need to post complete code you need to post code that actually reflects your issue. Right now you did not and you expect people to magically debug your program without seeing it, which is not particularly respectful.

**Reply by SadE54** [reply]
I m in discussion part, I did not want someone trying to find any bug, I
just want to understand and be guided a little. That being said, it s
probably not a bug, but an interaction with other imgui lib. I m a new
imgui user, I don't know all the mechanisms I m just learning...  Mais
merci le message est passé 😏

**Comment by rokups** [comment]
@SadE54 if you could provide a snippet we could paste to example application and reproduce the issue we would be glad to investigate. Currently provided code is not really in such shape to allow us to reproduce the issue.

**Reply by SadE54** [reply]
Yeah I understand that. But here the problem is that I cannot find a way to reproduce the behavior. My app is linked with another imgui lib and i cannot decently ask someone to get the same setup than mine. This part of the code is really obvious , and that almost a copy/paste I did before. The problem is probably coming from other imgui lib interaction I have to spot. And if I manage to spot it , I have the feeling that I won't need help anymore . It's like squaring the circle 🤔
